src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,201 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "programator12341";
+    private static final String DEFAULT_PASSWORD = "passatic123";
+    private static final int DEFAULT_AGE = 19;
+    private static final int AGE_LIMIT = 18;
+    private static final int LOGIN_MIN_LENGTH = 14;
+    private static final int PASSWORD_MIN_LENGTH = 10;
+    private static final String PATTERN_FOR_PASSWORD_LOGIN = "[A-Za-z0-9]*";
+    private static final String SHORT_PASSWORD = "short";
+    private static final String SHORT_LOGIN = "short";
+    private static final String MISMATCH_LOGIN = "--------------";
+    private static final String MISMATCH_PASSWORD = "%$#%@**(*&*(";
+
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private final Storage storage = new Storage();
+    private final User defaultUser = new User();
+    private final User defaultCopyUser = new User();
+    private final User nullUser = new User();
+    private final User nullPasswordUser = new User();
+    private final User nullLoginUser = new User();
+    private final User nullAgeUser = new User();
+    private final User shortPasswordUser = new User();
+    private final User shortLoginUser = new User();
+    private final User misMathcPatternUser = new User();
+
+    @BeforeEach
+    void setUp() {
+        defaultUser.setAge(18);
+        defaultUser.setLogin(DEFAULT_LOGIN);
+        defaultUser.setPassword(DEFAULT_PASSWORD);
+
+        defaultCopyUser.setAge(DEFAULT_AGE);
+        defaultCopyUser.setLogin(DEFAULT_LOGIN);
+        defaultCopyUser.setPassword(DEFAULT_PASSWORD);
+
+        nullUser.setAge(null);
+        nullUser.setLogin(null);
+        nullUser.setPassword(null);
+
+        nullLoginUser.setPassword(DEFAULT_PASSWORD);
+        nullLoginUser.setLogin(null);
+        nullLoginUser.setAge(DEFAULT_AGE);
+
+        nullPasswordUser.setAge(DEFAULT_AGE);
+        nullPasswordUser.setLogin(DEFAULT_LOGIN);
+        nullPasswordUser.setPassword(null);
+
+        nullAgeUser.setLogin(DEFAULT_LOGIN);
+        nullAgeUser.setPassword(DEFAULT_PASSWORD);
+
+        shortLoginUser.setLogin(SHORT_LOGIN);
+        shortLoginUser.setPassword(DEFAULT_PASSWORD);
+        shortLoginUser.setAge(DEFAULT_AGE);
+
+        shortPasswordUser.setLogin(DEFAULT_LOGIN);
+        shortPasswordUser.setPassword(SHORT_PASSWORD);
+        shortPasswordUser.setAge(DEFAULT_AGE);
+
+        misMathcPatternUser.setPassword(MISMATCH_PASSWORD);
+        misMathcPatternUser.setLogin(MISMATCH_LOGIN);
+        misMathcPatternUser.setAge(DEFAULT_AGE);
+    }
+
+    @AfterEach
+    void afterEach() {
+        storage.people.clear();
+    }
+
+    @Test
+    void current_Login_Is_Not_Exists_In_Storage_Is_Ok() {
```

Comment:
```suggestion
    void register_loginIsNotExistsInStorageIs_ok() {
```
please re-read the [rules ](https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea)for naming tests. fix it in all places